api/v1alpha4/azuremachine_validation.go

```diff
@@ -132,27 +130,23 @@ func ValidateOSDisk(osDisk OSDisk, fieldPath *field.Path) field.ErrorList {
 		allErrs = append(allErrs, field.Required(fieldPath.Child("OSType"), "the OS type cannot be empty"))
 	}
 
-	allErrs = append(allErrs, validateStorageAccountType(osDisk.ManagedDisk.StorageAccountType, fieldPath)...)
-
 	allErrs = append(allErrs, validateCachingType(osDisk.CachingType, fieldPath)...)
 
-	if errs := ValidateManagedDisk(osDisk.ManagedDisk, osDisk.ManagedDisk, fieldPath.Child("managedDisk")); len(errs) > 0 {
-		allErrs = append(allErrs, errs...)
+	if osDisk.ManagedDisk != nil {
+		if errs := validateManagedDisk(osDisk.ManagedDisk, fieldPath.Child("managedDisk")); len(errs) > 0 {
+			allErrs = append(allErrs, errs...)
+		}
 	}
 
-	if err := validateDiffDiskSetings(osDisk.DiffDiskSettings, fieldPath.Child("diffDiskSettings")); err != nil {
-		allErrs = append(allErrs, err)
+	if osDisk.ManagedDisk != nil && osDisk.ManagedDisk.StorageAccountType == string(compute.StorageAccountTypesUltraSSDLRS) {
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("managedDisks").Child("storageAccountType"), osDisk.ManagedDisk.StorageAccountType, "UltraSSD_LRS can only be used with data disks, it cannot be used with OS Disks"))
 	}
 
-	if osDisk.DiffDiskSettings != nil && osDisk.DiffDiskSettings.Option == string(compute.Local) && osDisk.ManagedDisk.StorageAccountType != "Standard_LRS" {
-		allErrs = append(allErrs, field.Invalid(
-			fieldPath.Child("managedDisks").Child("storageAccountType"),
-			osDisk.ManagedDisk.StorageAccountType,
-			"storageAccountType must be Standard_LRS when diffDiskSettings.option is 'Local'",
-		))
+	if err := validateDiffDiskSettings(osDisk.DiffDiskSettings, fieldPath.Child("diffDiskSettings")); err != nil {
+		allErrs = append(allErrs, err)
 	}
 
-	if osDisk.DiffDiskSettings != nil && osDisk.DiffDiskSettings.Option == string(compute.Local) && osDisk.ManagedDisk.DiskEncryptionSet != nil {
+	if osDisk.DiffDiskSettings != nil && osDisk.DiffDiskSettings.Option == string(compute.Local) && osDisk.ManagedDisk != nil && osDisk.ManagedDisk.DiskEncryptionSet != nil {
```

Comment:
```suggestion
	if osDisk.DiffDiskSettings != nil && osDisk.ManagedDisk != nil && osDisk.ManagedDisk.DiskEncryptionSet != nil {
```
if the kubebuilder comment below is valid.